### PR TITLE
Coinbase Pay Integration

### PR DESCRIPTION
**Pull request by nkrulikovsky**
## Description
Coinbase pay integration button

- Integrate coinbase pay button (user can add crypto to their wallet with Coinbase Pay Button)
- We can test on `https://saddlefinance.on.fleek.co/#/coinbase-pay-test` after merge
- Coinbase pay don't support all domain so we can test coinbase pay on prod domain

## Context
https://github.com/saddle-finance/tickets/issues/319

### CODE REVIEW

File: src/components/CoinbasePayBtn.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useRef, useState } from "react"
+import CnButtonImg from "../assets/button-cbPay-compact-addCrypto.png"
+import { initOnRamp } from "@coinbase/cbpay-js"
+import { styled } from "@mui/material"
+
+const CnPayButton = styled("button")(() => ({
```
Comment (ug02fast) [inline src/components/CoinbasePayBtn.tsx:6]: rename to CoinbasePayButton or StyledCoinbaseBtn or something similar. What is Cn?

File: src/components/CoinbasePayBtn.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+  }, [])
+
+  const handleClick = () => {
+    if (cbInstance?.current) {
+      cbInstance?.current.open()
+    }
+  }
+  return (
+    <CnPayButton
```
Comment (ug02fast) [inline src/components/CoinbasePayBtn.tsx:64]: Name is confusing to me.

File: tsconfig.json [hunk 3]
```diff
@@ -1,11 +1,7 @@
 {
   "compilerOptions": {
-    "target": "es5",
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "esnext"
-    ],
+    "target": "ES5",
```
Comment (ug02fast) [inline tsconfig.json:3]: is this necessary? can you revert this change and see if things are fine? sometimes you just need to restart TS Server on VSCode is lint errors come up.

File: src/pages/CoinbsePayTest.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import CoinbasePayBtn from "../components/CoinbasePayBtn"
+import { Container } from "@mui/material"
+import React from "react"
+
+export default function CoinbsePayTest(): JSX.Element {
```
Comment (ug02fast) [inline src/pages/CoinbsePayTest.tsx:5]: CoinbasePayTest rename

File: package.json [hunk 5]
```diff
@@ -117,18 +118,11 @@
...
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  },
+  "browserslist": [
```
Comment (ug02fast) [inline package.json:121]: what does browserslist do? does rest of app work fine after this change?

**Review by ug02fast — COMMENTED** [review]
Before merging, curious about a few things.

File: src/components/PoolInfoCard.tsx [hunk 6]
```diff
@@ -172,6 +174,11 @@ function PoolInfoCard({ data }: Props): ReactElement | null {
           </Typography>
         </InfoItem>
 
+        <Typography>{formattedData.futureA}</Typography>
+        <Typography>
+          {formattedData.futureATime.toLocaleDateString()}
```
Comment (ug02fast) [inline src/components/PoolInfoCard.tsx:179]: why is this addition necessary to test wallet?

File: src/components/CoinbasePayBtn.tsx (hunk 1 above)
Comment (nkrulikovsky) [reply src/components/CoinbasePayBtn.tsx:6]: was going to type type `CbPayButton` referring the `coinbase pay button`

File: tsconfig.json (hunk 3 above)
Comment (nkrulikovsky) [reply tsconfig.json:3]: Same with origin. only changed the text transform, but I'll change to lowercase.

Comment (ug02fast) [reply tsconfig.json:3]: Thanks

File: src/components/PoolInfoCard.tsx (hunk 6 above)
Comment (nkrulikovsky) [reply src/components/PoolInfoCard.tsx:179]: My mistake.
This file is not for this PR.
I think it was added while merge with master. will remove.

Comment (nkrulikovsky) [reply src/components/PoolInfoCard.tsx]: Removed!

File: package.json (hunk 5 above)
Comment (nkrulikovsky) [reply package.json:121]: > The browserslist configuration controls the outputted JavaScript so that the emitted code will be compatible with the browsers specified.  (https://create-react-app.dev/docs/supported-browsers-features/#configuring-supported-browsers)

**Review by ug02fast — APPROVED** [review]
